lib/active_merchant/billing/gateways/nmi.rb

```diff
@@ -297,9 +299,9 @@ def exp_date(payment_method)
 
       def commit(action, params)
         params[action == 'add_customer' ? :customer_vault : :type] = action
-        params[:username] = @options[:login]
-        params[:password] = @options[:password]
-
+        params[:username] = @options[:login]  unless @options[:login].nil?
+        params[:password] = @options[:password]  unless @options[:password].nil?
+        params[:security_key] = @options[:security_key] unless @options[:security_key].nil?
```

Comment:
❓ Question:

Why are you sending the security key on the request body ?, is already present on the Authorization header

Reply:
Hi @Heavyblade the corret way is send this in the request body, I will remove the security_key from the authorization header, Thx.